### DISCUSSION TITLE
Authentication (Token check) before file upload

### DISCUSSION THREAD

**Discussion by FirasKallel** [fastapi/fastapi#8850] [Questions] [answered]
How can i check ouath2 token before the file is uploaded?

In the current configuration the file is uploaded and then the dependency is called to check the token.
Instead of wasting resources untill the file is completely uploaded (especially if it's a big file) I would like to check the token before it completes the upload

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
def upload_file(
        user_id=Depends(verify_access_token),
        ufiles: UploadFile = File(...),
):
    return {"Hello": "World"}
```

**Comment by insomnes** [comment, marked as answer]
I believe that some kind of auth [middleware](https://fastapi.tiangolo.com/tutorial/middleware/) can help you here.

**Comment by bezaca** [comment]
Have you tried using an asynchronous function for this case ?

**Comment by MohamedAmineOuali** [comment]
> I believe that some kind of auth [middleware](https://fastapi.tiangolo.com/tutorial/middleware/) can help you here.

I don't think middleware could be used in this case. The middleware is applied to all routes. I don't think, it is good practice to filter the routes in middleware.

**Comment by insomnes** [comment]
> I don't think middleware could be used in this case. The middleware is applied to all routes. I don't think, it is good practice to filter the routes in middleware.

You can submount another app with middleware for specific routes.

**Comment by jziolkowski** [comment]
But does it really depend on the actual endpoint/middleware? Isn't the file upload being the part of the request? Therefore only if the file is uploaded, the request is "complete" and sent to the server? I think that this is part of how HTTP works, not fastapi itself.
